packages/mds-ingest-service/repository/index.ts

```diff
@@ -170,6 +170,12 @@ class IngestReadWriteRepository extends ReadWriteRepository {
         .getRepository(EventEntity)
         .createQueryBuilder('events')
         .innerJoin(qb => qb.from(DeviceEntity, 'd'), 'devices', 'devices.device_id = events.device_id')
+        .leftJoinAndMapOne(
+          'events.annotation',
+          EventAnnotationEntity,
+          'annotation',
+          'annotation.timestamp = events.timestamp AND annotation.device_id = events.device_id'
```

Comment:
I asked @terloon to open a new ticket to refactor the DB relationship, because this kind of join isn't necessary. 
I'm going to freeze approval on this PR until the `annotation` table has some kind of `event_id` to be joined on.

Reply:
For context - I agree, I'd love an `event_id` and had discussed that with @drtyh2o when planning the event annotations table initially. We've been indexing events by device_id + timestamp for a while now so the thinking was to continue with that for now and add a proper `event_id` as separate work at some point in the future.

Will this be expensive without `event_id`? I'm excited to add `event_id` sooner than later but if this works as is and `event_id` is a big refactor it would be a shame to block this OA feature. cc @avatarneil for any context on why we don't already have `event_id` and how hard it would be to add.

Reply:
Honestly, the context around why we don't have it is that it just didn't exist in 'user-facing' MDS ever (for basically no reason), and we never had a *need* to add it (and we had *needs* for plenty of other things that took up cycles 😄 ). I 100% think we should add it, and add it now, but note that _someday_ we'll probably have this added as a first-class notion in MDS and not want it auto-generated.